src/esp/assets/ResourceManager.h

```diff
@@ -774,6 +773,44 @@ class ResourceManager {
    */
   void buildPrimitiveAssetData(const std::string& primTemplateHandle);
 
+  /**
+   * @brief this will build a Phong @ref Magnum::Trade::MaterialData using
+   * default attributes from deprecated/removed esp::gfx::PhongMaterialData,
+   * if values are not specified.
+   * @param amb The ambient color
+   * @param diff The diffuse color
+   * @param spec The specular color
+   * @param shininess The shininess coefficient/exponent
+   * @return The new phong color
+   */
+  Mn::Trade::MaterialData buildDefaultPhongMaterial(
+      Mn::Color4 amb = Mn::Color4{0.1},
+      Mn::Color4 diff = Mn::Color4{0.7 * 0.175},
+      Mn::Color4 spec = Mn::Color4{0.2 * 0.175},
+      float shininess = 80.0f);
+
+  /**
+   * @brief Define and set user-defined attributes for the passed
+   * @ref Magnum::Trade::MaterialData.
+   * @param material The material to initialize with the expected
+   * Habitat-specific user-defined attributes.
+   * @param shaderTypeToUse What shader to use to render the objects with this
+   * material. May not be the same as the material type.
+   * @param hasVertObjID Whether or not the material has vertex-based object ids
+   * for semantics.
+   * @param hasTxtrObjID Whether or not the material has texture-based object
+   * ids for semantics.
+   * @param txtrIdx The absolute index in the @ref textures_ store for the semantic
+   * annotation texture.
+   * @return the updated material
+   */
+  Mn::Trade::MaterialData setDefaultMaterialUserAttributes(
```

Comment:
Nit: `Default` is misleading. `setMaterialUserAttributes` might be a better name here.

Reply:
They are default because they are expected to be present in all materials (the "default user attributes". How about "setMaterialDefaultUserAttributes" ?